tests/test_empiricalVariance.py

```diff
@@ -0,0 +1,149 @@
+#
+# LSST Data Management System
+# Copyright 2018 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import numpy as np
+import unittest
+
+import lsst.utils.tests
+
+from lsst.daf.base import PropertyList
+from lsst.afw.cameraGeom import Detector, SCIENCE, Orientation, TransformMap, CameraSys
+from lsst.afw.table import AmpInfoCatalog, AmpInfoTable
+from lsst.afw.geom import Point2I, Extent2I, Box2I, Extent2D
+from lsst.afw.image import ExposureF, VisitInfo
+from lsst.pipe.base import Struct
+
+from lsst.ip.isr.isrTask import IsrTask
+
+
+def makeAmplifier(catalog, name, bbox, rawImageBox, overscanBox, gain, readNoise, saturation):
+    amp = catalog.addNew()
+    amp.setName(name)
+    amp.setBBox(bbox)
+    amp.setRawDataBBox(rawImageBox)
+    amp.setRawHorizontalOverscanBBox(overscanBox)
+    amp.setHasRawInfo(True)
+    amp.setGain(gain)
+    amp.setReadNoise(readNoise)
+    amp.setSaturation(saturation)
+    amp.setSuspectLevel(np.nan)
+    return amp
+
+class EmpiricalVarianceTestCast(lsst.utils.tests.TestCase):
+    def setUp(self):
+        """Constructs a CCD with two amplifiers and prepares for ISR"""
```

Comment:
Could you use a real image from some testdata rather than building a fake one here? I'd be more inclined to believe the test result for a real-life exposure than for this minimalistic fake one.

Reply:
obs_decam tests include running `processCcd.py` on some real DECam data, and ci_hsc does similarly with real HSC data. Both ran fine under Jenkins.

I think leaving the tests on real data to other packages is the right balance: it keeps the size of this code repo small, and using simulated data means we can isolate the features being tested, and we know the exact right answer.